File: 프로그래머스/lv2/오하늘/124 나라의 숫자.py
# 3진법이라는 과정 보고 재 풀이
def solution(n):
    answer = ''
    arr = ['1','2','4'] # 3으로 나눴을때 인덱스가 나와야함
    while n > 0 :
        n=n-1 # -1을 해주면 각 0,1,2가 나옴
        answer = arr[n%3]+answer
        n//=3
    
    return answer

File: 프로그래머스/lv2/김현진/124 나라의 숫자.cpp
//1을 빼는 아이디어.. 정확히 이해가 안돼요..

#include <string>
#include <vector>
#include<algorithm>

using namespace std;

string solution(int n) {
    string answer = "";
    while(n>0)
    {
        if(n%3==0)
        {
            answer += "4";
            n = n/3 -1;
        }
        else if(n%3==1)
        {
            answer += "1";
            n/=3;
        }
        else if(n%3==2)
        {
            answer += "2";
            n/=3;
        }
    }
    reverse(answer.begin(),answer.end());
    return answer;
}
